### ISSUE TITLE
OTPose: Occlusion-Aware Transformer for Pose Estimation in Sparsely-Labeled Videos

### ISSUE THREAD

**Issue by Songinpyo** [Songinpyo/Paper-Record#16]
## ❓ Why → 해당 논문의 필요성, abstract
CNN based multi person pose estimation in video는 각 keypoint 간의 관계를 frame에 따라 나타내는 기본적인 아이디어를 따른다.
하지만, occlusion과 motion blur가 이를 저해한다.
또한 CNN은 근본적으로 receptive field가 제한적이다. 이를 극복하기 위해 network를 deep하게 만든다면 모든 pixel에 대해서 long-range dependency가 증가해 역효과를 부를 수 있다.

OTPose는 Frames의 여러 조합을 이용해서 overall joint movement를 추적하며, Semi-supervised 방식으로 occlusion aware heatmaps를 생성할 수 있게 하는 occlusion attention mask를 제안한다.

또한, Transformer 구조를 채택하여 temporal relationship과 keypoint wise attention을 각 frame에 대해서 연결지어 target frame의 정확한 estimation이 가능하다.

## 💡 Key insights → 해당 논문의 핵심 아이디어
![image](https://user-images.githubusercontent.com/104220612/187366644-617b32bb-23cb-47bd-9062-099a1ac38d1f.png)
5frames를 선택하여 Frames를 조합한 Keypoint Flow에 5frames를 모두 더한 Channel-wise summed Pose를 Element-wise product하여 occlusion Attention Mask를 생성한다.

![image](https://user-images.githubusercontent.com/104220612/187366886-5d40db9e-8b4a-43f2-b552-5e276217511b.png)
위에서 구한 Occlusion Attention Mask와 Keypoint flow, Occlusion-aware heatmap을 transformer의 encoder를 이용하여 Temporal encoding하고 이후 Deformable convolution을 통해 최종 heatmap을 추정한다.

📌 **Occlusion Encoder**
![image](https://user-images.githubusercontent.com/104220612/187367184-3ac976aa-b499-40f1-a76d-e478e4b4fca1.png)
Total keypoint flow를 Occlusion Encoder에 넣어 구한 Occlusion aware heatmap을 기존의 GT인 G와 Mask M을 더한 Pseudo label과 비교하여 occlusion loss를 구한다.

📌 **Temporal encoding**
![image](https://user-images.githubusercontent.com/104220612/187367569-b60b190c-05eb-4f54-9c44-ff0074b28766.png)
Transformer encoder를 이용해서 과거와 미래에 대해 temporal encoding 된 2개의 heatmaps를 구한다.

Transformer에 사용하기 위해 이전 Occlusion Attention Mask, Occlusion-aware heatmap, Keypoint flow를 patch embedding 하여야 하는데 이는 [vivit](https://openaccess.thecvf.com/content/ICCV2021/papers/Arnab_ViViT_A_Video_Vision_Transformer_ICCV_2021_paper.pdf) 논문의 방법을 택하였다.
vivit에 대해서는 차후 따로 다루도록 하겠다.

📌 **Loss function**
![image](https://user-images.githubusercontent.com/104220612/187368545-5afb439f-ee21-4d8a-9a74-21d21f2ba7ed.png)
loss는 앞서 구한 occlusion loss, GT와 pred를 비교한 pred loss 두 개로 구성되어진다.
각각 1/17, 8/17 이 곱해진다.
N_j = 17, N_f = 8

## 🔗 Reference URL → 논문 링크
[OTPose: Occlusion-Aware Transformer for Pose Estimation in Sparsely-Labeled Videos](https://arxiv.org/abs/2207.09725)
